src/Symfony/Component/Translation/IdentityTranslator.php

```diff
@@ -25,16 +25,18 @@ class IdentityTranslator implements LegacyTranslatorInterface, TranslatorInterfa
     use TranslatorTrait;
 
     private $selector;
+    private $behaveSilently;
 
     /**
      * @param MessageSelector|null $selector The message selector for pluralization
      */
-    public function __construct(MessageSelector $selector = null)
+    public function __construct(MessageSelector $selector = null, bool $behaveSilently = false)
     {
         $this->selector = $selector;
+        $this->behaveSilently = $behaveSilently;
 
-        if (__CLASS__ !== \get_class($this)) {
-            @trigger_error(sprintf('Calling "%s()" is deprecated since Symfony 4.2.', __METHOD__), E_USER_DEPRECATED);
+        if (__CLASS__ !== \get_class($this) && null !== $this->selector) {
+            @trigger_error(sprintf('Passing %s in "%s()" is deprecated since Symfony 4.2.', MessageSelector::class, __METHOD__), E_USER_DEPRECATED);
```

Comment:
this misses a plan for v5.0: how is one supposed to not pass a messge selector then?
instead, the new argument should be passed first, and the logic here should be able to deal with hydribe types and trigger a deprecation notice when a message selector is passed